pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -1050,10 +853,10 @@ impl<T: AsRef<DeltaLayerInner>> ValueRef<T> {
     }
 }
 
-struct Adapter<T: AsRef<DeltaLayerInner>>(T);
+struct Adapter<'a>(&'a DeltaLayerInner);
 
-impl<T: AsRef<DeltaLayerInner>> BlockReader for Adapter<T> {
+impl BlockReader for Adapter<'_> {
     fn read_blk(&self, blknum: u32) -> Result<BlockLease, std::io::Error> {
-        self.0.as_ref().file.read_blk(blknum)
+        self.0.file.read_blk(blknum)
     }
 }
```

Comment:
These changes have been cleared with Arpad, expecting minor conflicts here, but very small.

Reply:
Took me a while to figure them out, but now pushed.